website/content/en/guides/developer/debugging/debugging.md

```diff
@@ -0,0 +1,760 @@
+---
+title: A guide for debugging Vector pipelines
+short: Debugging Vector
+description: Step by step guide for debugging pipelines
+tags: [ "dev", "debugging", "guides", "guide" ]
+author_github: https://github.com/pront
+domain: dev
+aliases: [ "/docs/guides/developer/config-autocompletion.md" ]
+weight: 1
+---
+
+# Debugging Guide
+
+This guide will describe an imaginary use case and how to progressively resolve issues using dev tools and Vector commands.
+
+Imagine we have two sources, and we want to assign group IDs to Vector logs based on the source. Finally, we want to send those modified
+logs to an HTTP server for further processing and/or storage.
+
+In the following sections we will examine the tools we have at our disposal.
+
+<img src="/img/guides/debugging-meme.png"  alt="debugging-meme" width="400"/>
+
+## Vector Tools
+
+### Vector top
+
+Use [vector top](https://vector.dev/docs/reference/cli/#top) to display topology and metrics in the console.
+For example, you can inspect how many events are produced by sources vs how many events reach the sinks.
+
+Command:
+
+```shell
+vector top
+```
+
+Screen:
+
+<img src="/img/guides/vector-top.png"  alt="top-screen"/>
+
+### The internal metrics source
+
+The [internal_metrics](https://vector.dev/docs/reference/configuration/sources/internal_metrics/) source can be used to inspect component
```

Comment:
I'd recommend showing how they can be piped to a `prometheus_exporter` sink and scraped with `curl` to examine. Otherwise readers may not know what to do with the `internal_metrics` source.